### PR TITLE
Mark AOT backward stack comments in print_readable

### LINKED ISSUE #166135: Stack comments in aot_joint_graph should indicate if it's backwards
### 🐛 Describe the bug

Here is a funny fragment of code I was looking at the other day from the aot_joint_graph print readable:

```
         # File: REDACTED.py:150 in forward, code: input = input.to(dtype=torch.float32)
        convert_element_type_XXX: "bf16[..., 16]cuda:0" = torch.ops.prims.convert_element_type.default(add_XXX, torch.bfloat16)
```

Bug? No. This is actually just the backwards pass. But it's confusing that the comment doesn't say so. We should!

cc @chauhang @penguinwu @zou3519 @bdhirsh @yushangdi 

### Versions

main

### PR DESCRIPTION
**Pull request by jansel**
Stack from [ghstack](https://github.com/ezyang/ghstack/tree/0.15.0) (oldest at bottom):
* __->__ #185067

AOT joint graphs reuse forward stack traces on nodes traced while running backward. Those nodes already carry autograd_backward metadata, but FX readable printing rendered the stack summary exactly like a forward node, so a backward cast could look like the source forward cast.

Use the existing autograd_backward metadata when formatting readable stack comments and prefix those comments with "Backward of forward node:". This keeps the logic in the shared FX readable formatter used by aot_joint_graph instead of duplicating stack formatting in AOTAutograd. Add a focused AOT regression that captures the joint graph before partitioning and checks the forward and backward cast comments.

Fixes #166135
Generated by my agent

Test Plan:
- python inline aot_module_simplified joint-graph repro: passed
- python test/functorch/test_aotdispatch.py -k test_aot_joint_print_readable_marks_backward_stack_trace with a temporary torchvision stub: passed, 1 test
- python test/functorch/test_aotdispatch.py -k test_aot_joint_print_readable_marks_backward_stack_trace: blocked before collection by local torchvision::nms import error
- lintrunner -a: passed
- git diff --check: passed

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @kadeng @chauhang @amjames @jataylo @azahed98

### CODE REVIEW

**Comment by jansel** [comment]
@claude review these changes

**Review by SherlockNoMad — APPROVED** [review]

**Review by mlazos — APPROVED** [review]

**Review by eellison — COMMENTED** [review]
address failing test

**Comment by jansel** [comment]
My agent says I addressed the failing test by updating the affected test/higher_order_ops/test_invoke_subgraph.py inline expectation after the new backward stack comment is stripped. The targeted test_bwd_partitioning test now passes locally, and the PR has been resubmitted.

**Comment by jansel** [comment]
My agent says I addressed the new Dr.CI failures by updating the expected readable-graph outputs in test_fwd_loss_bwd, test_subclasses, and test_control_flow. These failures were caused by normalize_gm stripping the new backward stack-comment text while preserving the blank separator line; the targeted failing tests now pass locally.

**Comment by jansel** [comment]
My agent says I addressed the new Dr.CI failure in test_autograd_grad_single_tensor by updating its expected readable-graph output for the blank separator line left after normalize_gm strips the new backward stack-comment text. The targeted test now passes locally, and the PR has been resubmitted.

**Comment by jansel** [comment]
My agent says I addressed the latest Dr.CI failures in test_autograd_grad_with_ingraph_nn_parameter by updating its expected readable-graph output for the blank separator line left after normalize_gm strips the new backward stack-comment text. The targeted failing test now passes locally, and the PR has been resubmitted.

**Comment by jansel** [comment]
My agent says I addressed the new Dr.CI failures in test_autograd_grad_with_unrelated_requires_grad_output by updating its expected readable-graph output for the blank separator line left after normalize_gm strips the new backward stack-comment text. The targeted failing test now passes locally, lint is clean, and the PR has been resubmitted.

**Comment by jansel** [comment]
My agent says I addressed the latest Dr.CI test_tensor_backward_accumulates_grads failure by updating that expected readable-graph output, and also fixed the adjacent tensor_backward expectations with the same stripped-comment separator behavior. The local test_fwd_loss_bwd tensor_backward and autograd_grad sweeps pass, lint is clean, and the PR has been resubmitted. The remaining inductor benchmark failures appear to be eager/model-validation nondeterminism unrelated to this print_readable change.

**Comment by jansel** [comment]
My agent says I addressed the new Dr.CI failure in test_graph_deduplication.py::GraphDededuplicationTests::test_single_subgraph2 by updating the normalized readable-graph expectation for the blank separator line left after the backward stack-comment text is stripped. The targeted test and adjacent single_subgraph sweep pass locally, lint is clean, and the PR has been resubmitted.

**Comment by jansel** [comment]
My agent says I resubmitted after the revert fallout. The update adds the missed print_readable expected-output changes for the CUDA stream backward graph tests and test_leaf_function_no_return_value, and I reran those targeted tests plus lint.

**Comment by jansel** [comment]
@claude is this safe to reland? Give a confidence level.
